packages/authentication/src/__tests__/fixtures/services/basic-auth-user-service.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors} from '@loopback/rest';
+import {UserService} from '../../../services/user.service';
+import {UserProfile} from '../../../types';
+import {USER_REPO} from '../keys';
+import {BasicAuthenticationStrategyCredentials} from '../strategies/basic-strategy';
+import {User} from '../users/user';
+import {UserRepository} from '../users/user.repository';
+
+export class BasicAuthenticationUserService
+  implements UserService<User, BasicAuthenticationStrategyCredentials> {
+  constructor(
+    @inject(USER_REPO)
+    private userRepository: UserRepository,
+  ) {}
+
+  async verifyCredentials(
+    credentials: BasicAuthenticationStrategyCredentials,
+  ): Promise<User> {
+    if (!credentials) {
+      throw new HttpErrors.Unauthorized(`'credentials' is null`);
+    } //if
+
+    if (!credentials.email) {
+      throw new HttpErrors.Unauthorized(`'credentials.email' is null`);
+    } //if
+
+    if (!credentials.password) {
+      throw new HttpErrors.Unauthorized(`'credentials.password' is null`);
+    } //if
+
+    const foundUser = this.userRepository.find(
+      credentials.email,
+      credentials.password,
+    );
+    if (!foundUser) {
+      throw new HttpErrors['Unauthorized'](
+        `User with email ${credentials.email} not found.`,
+      );
+    }
+
+    if (credentials.password !== foundUser.password) {
+      throw new HttpErrors.Unauthorized('The password is not correct.');
+    } //if
+
+    return foundUser;
+  }
+
+  convertToUserProfile(user: User): UserProfile {
+    if (!user) {
+      throw new HttpErrors.Unauthorized(`'user' is null`);
```

Comment:
I personally like adding `//if` and `//endif` and `//try` and `//catch` next to the `}` in complex nested blocks in order to make my code easier to read...but I will remove the `//if` comments in this piece of code because the code isn't complex.

Reply:
Oh ok I am ok with either using the comment or not.

Reply:
Modern IDE such as VSCode can help find matching `{}` or `()`. Adding such comments is not helpful for other developers as `if` can be nested too.